### ISSUE TITLE
Add advanced Ground boundary conditions to boundarycondition.py

### ISSUE THREAD

**Issue by chriswmackey** [ladybug-tools/honeybee-energy#518]
Hi @chriswmackey ,

I'm trying to extend the boundary construction library to include the GroundSlab and GroundBasement items and, if I understand correctly, it's implemented in a similar way as this. If you could give me some basic directions, I think it could be helpful to have this set of options in the library of additional boundary conditions.

So far, I am testing one of the BC to find my way around it (_GroundSlabPreprocessorAverage_). Although I infer I have broken any possible code rule here, I'll share what I'm trying so far.

1. Add another class to boundarycondition.py, replicating the adiabatic one.
![image](https://user-images.githubusercontent.com/25599446/103641951-c969d180-4f52-11eb-92a0-97852c85101d.png)

2. Add another set of boundary condition to _extend_honeybee.py
![image](https://user-images.githubusercontent.com/25599446/103642057-ec948100-4f52-11eb-9b4c-2e75b43b9e96.png)

3. Test in gh, where it throws the error _1. Solution exception:global name 'GroundSlabPreprocessorAverage' is not defined_ both in the case where I define the new BC and if try a regular one.
![image](https://user-images.githubusercontent.com/25599446/103642262-31201c80-4f53-11eb-9e27-3f47216bfd54.png)

I have to say that I have little knowledge of what I'm doing, and I have mostly copy pasted the adiabatic objects and replaced certain text strings. 😅 

Thanks, and happy new year!
Rafael

_Originally posted by @rafael-a in https://github.com/ladybug-tools/honeybee-energy/issues/407#issuecomment-754589653_

**Comment by chriswmackey** [comment]
Hi @rafael-a ,

Happy new year to you too.  I moved your comment to a new issue as the other issue is different and will likely be closed before the issue you have here gets addressed.

I say this because (I think) these advanced ground boundary conditions require the running of preprocessor methods before the EnergyPlus run, correct?  So they're going to involve setting several more parameters on a boundary condition object class than you have for something like Adiabatic or OtherSideCoefficients.

The reason why the Grasshopper component fails is because, if you are adding a new type of boundary condition, you need to plug in a Boundary Condition object instance instead of a string.   However, this is a bit beside the point since I would recommend doing your tests in pure Python and not in the Grasshopper plugin while you are adding a feature to honeybee-energy.  This honeybee-energy repo here is completely independent of Grasshopper and getting things to work here and in 2 other repos (honeybee-schema and honeybee-openstudio-gem) is a prerequisite before we can expose things on the Grasshopper interface.

**Comment by rafael-a** [comment]
Hi @chriswmackey ,

thanks for bringing some clarity to this issue. I understand a bit better now the process involved in the new LadybugTools.

Correct, these boundary conditions require running the preprocessors, either the basement or slab one, or both. In addition to the boundary conditions defined on the geometry, this requires a few other additional E+ objects that launch the Preprocessors - in the past I had added them via the _additional strings_ route.

I'll try the pure python route then, and see if I manage to get somewhere :)
